### PR TITLE
Urban behaviors rework

### PR DESCRIPTION
**Pull request by Badulundabad**
- reworked assimilation behavior and component. Now it doesn't turn empire men into vampires, notables in all settlements are humans;
- removed excess extensions of settlement and hero classes;
- changed wanderer behavior to avoid cases where vampire companion is still sitting in empire town and vice versa.

### CODE REVIEW

File: CSharpSourceCode/CampaignSupport/Assimilation/AssimilationCampaignBehavior.cs [hunk 1]
```diff
@@ -13,18 +11,26 @@ public class AssimilationCampaignBehavior : CampaignBehaviorBase
         public override void RegisterEvents()
         {
             CampaignEvents.OnGameLoadFinishedEvent.AddNonSerializedListener(this, OnGameLoadFinishedEvent);
```
Comment (hunharibo) [inline CSharpSourceCode/CampaignSupport/Assimilation/AssimilationCampaignBehavior.cs:13]: I would suggest to use the OnSessionLaunched event instead. I think GameLoadFinished only fires when loading a savegame, not when starting a new game, and my guess is that you want both.

File: CSharpSourceCode/CampaignSupport/Assimilation/AssimilationComponent.cs [hunk 2]
```diff
@@ -1,284 +1,100 @@
...
-                    newNotable.SetPersonalRelation(otherHero, relation);
-                }
-            }
-            if (victim.Issue != null)
+            var outriders = settlement.Notables.Where(n => n.Culture != _settlement.Owner.Culture);
+            if (outriders.Count() > 0)
             {
-                Campaign.Current.IssueManager.ChangeIssueOwner(victim.Issue, newNotable);
+                var outrider = outriders.First();
+                outrider.Culture = _settlement.Owner.Culture;
```
Comment (hunharibo) [inline CSharpSourceCode/CampaignSupport/Assimilation/AssimilationComponent.cs:74]: I am not sure that setting the notable's culture directly changes the troops recruitable by them. Have you tested? If yes, that is very good news! I thought we would have to create new notables based on the xml settings in culture for it to work.,

Comment (Badulundabad) [reply CSharpSourceCode/CampaignSupport/Assimilation/AssimilationComponent.cs:74]: I've tested it a few months ago and it worked like a swiss watch

File: CSharpSourceCode/CampaignSupport/Assimilation/AssimilationCampaignBehavior.cs (hunk 1 above)
Comment (Badulundabad) [reply CSharpSourceCode/CampaignSupport/Assimilation/AssimilationCampaignBehavior.cs:13]: I don't add this component to every settlement at new game start, but to each settlement after its owner is changed and only if culture is different. After loading a save game I need to add already created components to their settlements. That's why I use OnGameLoadFinishedEvent

Comment (hunharibo) [reply CSharpSourceCode/CampaignSupport/Assimilation/AssimilationCampaignBehavior.cs:13]: OK, good enough :)

File: CSharpSourceCode/CampaignSupport/Assimilation/AssimilationComponent.cs (hunk 2 above)
Comment (hunharibo) [reply CSharpSourceCode/CampaignSupport/Assimilation/AssimilationComponent.cs:74]: Saw your picture as well on discord. Fine then, no more issues.

**Review by hunharibo — APPROVED** [review]
Approved after resolving the 2 comments. Nice work!

**Review by Z3rca — APPROVED** [review]
👍

**Pull request merged**